orangecontrib/educational/widgets/owkmeans.py

```diff
@@ -373,17 +354,17 @@ def replot(self):
         if self.data is None or not self.attr_x or not self.attr_y:
             return
 
-        if self.k_means.centroids_moved:
-            # when centroids moved during step
-            self.scatter.update_series(0, self.k_means.centroids)
-
-            if self.lines_to_centroids:
-                for i, c in enumerate(self.k_means.centroids):
-                    self.scatter.update_series(1 + i, list(chain.from_iterable(
-                        ([p[0], p[1]], [c[0], c[1]])
-                        for p in self.k_means.centroids_belonging_points[i])))
-        else:
+        km = self.k_means
+        if not km.centroids_moved:
             self.complete_replot()
+            return
+
+        self.scatter.update_series(0, km.centroids)
+        if self.lines_to_centroids:
+            for i, (c, pts) in enumerate(zip(km.centroids, km.centroids_belonging_points)):
```

Comment:
The loop is essentially iterating over both, `centroids` and `centroids_belonging_points`, so let us show it by zipping them. `i, (c, pts) in enumerate(zip(...))` may look a bit weird, but that's what the loop does here.